atlasdb-api/src/main/java/com/palantir/atlasdb/spi/AtlasDbFactory.java

```diff
@@ -69,13 +55,17 @@ default MERGED_CONFIG createMergedKeyValueServiceConfig(
      */
     KeyValueService createRawKeyValueService(
             MetricsManager metricsManager,
-            MERGED_CONFIG config,
+            KeyValueServiceConfig config,
             Refreshable<Optional<KeyValueServiceRuntimeConfig>> runtimeConfig,
             Optional<LeaderConfig> leaderConfig,
             Optional<String> namespace,
             LongSupplier freshTimestampSource,
             boolean initializeAsync);
 
+    DerivedConcurrencyConfig createDerivedConcurrencyConfig(KeyValueServiceConfig config,
```

Comment:
Some thoughts:
- Should this have a default impl? In most cases we just end up returning the KVS config object.
- Don't know if we want to have this be so explicit as a derived _concurrency_ config as opposed to some kind of derived snapshot config